src/hooks/useActiveDescendantListFocus/__tests__/useActiveDescendantListFocus.jest.js

```diff
@@ -5,32 +5,109 @@ import userEvent from "@testing-library/user-event";
 let element;
 const FIRST_ITEM_ID = "id-1";
 const SECOND_ITEM_ID = "id-2";
-const ITEM_IDS = [FIRST_ITEM_ID, SECOND_ITEM_ID];
-
-function renderHookForTest({ onItemClick = jest.fn(), isItemSelectable = () => true, isHorizontal = false }) {
+const THIRD_ITEM_ID = "id-3";
+const FOURTH_ITEM_ID = "id-4";
+const FIFTH_ITEM_ID = "id-5";
+const ITEM_IDS = [FIRST_ITEM_ID, SECOND_ITEM_ID, THIRD_ITEM_ID, FOURTH_ITEM_ID, FIFTH_ITEM_ID];
+
+function renderHookForTest({
+  onItemClick = jest.fn(),
+  keepOptionSelected = false,
+  isItemSelectable = () => true,
+  isHorizontal = false
+}) {
   element = document.createElement("div");
   element.tabIndex = -1; // some tests focus the element - a tabIndex value is required for updating the document.activeIndex value
   document.body.appendChild(element);
 
-  return renderHook(() =>
-    useActiveDescendantListFocus({
-      focusedElementRef: {
-        current: element
-      },
-      itemsIds: ITEM_IDS,
-      isItemSelectable: isItemSelectable,
-      onItemClick,
-      isHorizontalList: isHorizontal
-    })
-  );
+  const props = {
+    focusedElementRef: {
+      current: element
+    },
+    keepOptionSelected,
+    itemsIds: ITEM_IDS,
+    isItemSelectable: isItemSelectable,
+    onItemClick,
+    isHorizontalList: isHorizontal
+  };
+  return renderHook(argprops => useActiveDescendantListFocus({ ...props, ...argprops }));
 }
 
-function runListUnitTest(isHorizontal) {
+function runListUnitTest({ isHorizontal, keepOptionSelected }) {
   const moveForwardKey = isHorizontal ? "{arrowRight}" : "{arrowDown}";
   const oppositeMoveForwardKey = !isHorizontal ? "{arrowRight}" : "{arrowDown}";
+  const naturalIndex = keepOptionSelected ? 0 : undefined;
```

Comment:
its not natural focus because the natural index is always on the container. visualFocusIndex?

Reply:
and it's not good that the visual focus index before navigate to the combobox will be 0, it's affect the screen reader. it must be undefined :(

Reply:
I did not understand if it is the index before we focus on the wrapper or after, if it is before it's a problem. if it's after it is ok :)

Reply:
it's after focus

Reply:
added a test to make sure it starts as undefined